app/models/order.rb

```diff
@@ -111,6 +111,10 @@ class Order < ActiveRecord::Base
   scope :goodcity_orders, -> { where(detail_type: "GoodCity") }
   scope :shipments, -> { where(detail_type: "Shipment") }
 
+  def assign_state(status)
+    state = SHIPMENT_STATUS_MAP[status]
```

Comment:
Need a default state

```suggestion
    state = SHIPMENT_STATUS_MAP[status] || 'submitted'
```